public/app/features/alerting/unified/components/rule-viewer/tabs/extensions/RuleViewerExtension.tsx

```diff
@@ -5,7 +5,10 @@ import { withErrorBoundary } from '@grafana/ui';
 
 import { logError } from '../../../../Analytics';
 
-export interface RuleViewerExtensionProps {}
+export interface RuleViewerExtensionProps {
+  ruleUid: string;
```

Comment:
I introduced `ruleUid` as a prop to prevent consumers from relying on context, reducing coupling